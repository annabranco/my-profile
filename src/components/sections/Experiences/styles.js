import styled, { css } from 'styled-components';
import { rgba } from 'polished';
import { Sea } from '../../../assets/images';
import {
  colorBlueLight,
  fontTitle,
  colorRedDark,
  fontTitleAlt,
  colorBlueDark,
  fontSubtitle
} from '../../../styles/theme';

export const City = styled.span`
  font-size: 0.7rem;
  font-style: italic;

  @media all and (min-width: 768px) {
    font-size: 0.8rem;
  }
`;
City.displayName = 'City';

export const Company = styled.p`
  margin: 8px 0 0 15px;
  font-family: ${fontSubtitle};
  font-size: 0.8rem;

  @media all and (min-width: 768px) {
    font-size: 1rem;
  }
`;
Company.displayName = 'Company';

export const CountryFlag = styled.img`
  margin: 0 5px;
  vertical-align: middle;
`;
CountryFlag.displayName = 'Country Flag';

export const DateArea = styled.div`
  width: 12%;
`;
DateArea.displayName = 'Date Area';

export const Details = styled.p`
  margin-top: 10px;
`;
Details.displayName = 'Details';

export const DetailsArea = styled.div`
  margin-top: 10px;
  margin-left: 40px;
  transform: translate(800px, 0);
  transition: all 4s ease;

  ${({ visible }) =>
    visible &&
    css`
      width: 80%;
      transform: translate(0, 0);
    `}
  @media all and (min-width: 768px) {
    margin-top: 0;
  }

  ${({ modal }) =>
    modal &&
    css`
      background: rgba(255, 255, 255, 0.8);
      padding: 20px;
      border-radius: 5px;
    `}
`;
DetailsArea.displayName = 'Details Area';

export const ExperiencesArea = styled.div`
  position: relative;
  background-image: linear-gradient(
      rgba(230, 239, 251, 0.98),
      rgba(230, 239, 251, 0.93),
      rgba(189, 214, 246, 0.98)
    ),
    ${`url(${Sea})`};
  background-position: center;
  height: ${`${window.innerHeight}px`};

  @media all and (min-width: 768px) {
    height: auto;
  }
`;
ExperiencesArea.displayName = 'Experiences Area';

export const ExperiencesWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: flex-end;
  margin: 130px auto 0;
  padding-right: 5vw;
  height: 50%;

  @media all and (min-width: 768px) {
    margin: 0 auto;
    height: auto;
  }
`;
ExperiencesWrapper.displayName = 'Experiences Wrapper';

export const ExperienceItem = styled.div`
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
  margin: 10px 0;
  height: auto;
  width: 100%;
  opacity: 0;
  transition: opacity ease 3s;

  ${({ visible }) =>
    visible &&
    css`
      opacity: 1;
    `}

  @media all and (min-width: 768px) {
    margin: 20px 0;
  }
`;
ExperienceItem.displayName = 'Experience Item';

export const TextDate = styled.p`
  font-family: ${fontTitle};
  font-size: 0.8em;
  color: ${colorRedDark};
  text-align: center;

  @media all and (min-width: 768px) {
    font-size: 1.2em;
  }
`;
TextDate.displayName = 'TextDate';

export const Title = styled.h3`
  display: inline;
  margin-bottom: 20px;
  font-family: ${fontTitleAlt};
  font-size: 1em;
  color: ${colorBlueDark};

  @media all and (min-width: 768px) {
    font-size: 1.4em;
  }
`;
Title.displayName = 'Title';

export const VerticalBar = styled.div`
  position: absolute;
  left: 10%;
  border-right: 5px solid ${rgba(colorBlueLight, 0.3)};
  height: 100%;
  width: 0;
`;
VerticalBar.displayName = 'Vertical Bar';
