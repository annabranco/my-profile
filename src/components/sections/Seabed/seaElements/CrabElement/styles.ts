import styled, { css } from 'styled-components';

interface TheCrabProps extends React.ComponentPropsWithoutRef<'img'> {
  runnaway: boolean;
}

export const TheCrab = styled.img<TheCrabProps>`
  position: absolute;
  bottom: 30px;
  right: 50%;
  transform: translate(50%, 0);
  width: 60px;
  opacity: 0.5;
  filter: drop-shadow(0 5px 2px rgba(0, 0, 0, 0.2));
  transition: all 3s ease-in;

  ${({ runnaway }) =>
    runnaway &&
    css`
      right: -10vw;
      transform: none1;
      transition: all 4s ease-out;
    `}
`;
TheCrab.displayName = 'TheCrab';
