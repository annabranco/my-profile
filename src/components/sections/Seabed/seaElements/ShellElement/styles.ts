import styled, { keyframes } from 'styled-components';

export const TheShell = styled.img`
  width: 60px;
  opacity: 0.5;
  filter: drop-shadow(0 5px 2px rgba(0, 0, 0, 0.2));
`;
TheShell.displayName = 'TheShell';

export const ShellContainer = styled.div`
  z-index: 100;
  position: absolute;
  bottom: 30px;
  left: 15vw;
`;
ShellContainer.displayName = 'ShellContainer';

export const PearlShine = styled.div`
  position: absolute;
  top: 1px;
  left: 1px;
  height: 8px;
  width: 8px;
  border-radius: 50%;
  background: radial-gradient(circle at 100%, rgb(220, 220, 220), white);
  box-shadow: 0 0 10px 1px white;
`;
PearlShine.displayName = 'PearlShine';

const Shine = keyframes`
0% {
    box-shadow: 0 0 20px 4px yellow;
  }

  50% {
        box-shadow: none;
  }

  100% {
        box-shadow: 0 0 20px 4px yellow;
  }
`;

export const Pearl = styled.div`
  position: absolute;
  bottom: 14px;
  left: 23px;
  height: 15px;
  width: 15px;
  background: radial-gradient(
    circle at 100%,
    rgb(220, 220, 220),
    rgb(192, 192, 192) 50%,
    rgb(255, 255, 255) 75%,
    rgb(220, 220, 220) 15%
  );
  border-radius: 50%;
  filter: drop-shadow(0 5px 2px rgba(0, 0, 0, 0.2));
  border: 1px solid rgba(92, 92, 92, 0.3);

  animation-name: ${Shine};
  animation-duration: 2s;
  animation-timing-function: linear;
  animation-iteration-count: infinite;
  animation-fill-mode: forwards;
`;
Pearl.displayName = 'Pearl';
