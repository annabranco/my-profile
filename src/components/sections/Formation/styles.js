import styled, { css, keyframes } from 'styled-components';
import { rgba } from 'polished';
import {
  colorWhite,
  colorBlueLight,
  colorBlack,
  fontSubtitle,
  colorYellowBright,
  colorBlueWater,
  colorYellowDark,
  fontTitleAlt
} from '../../../styles/theme';
import { Title } from '../Experiences/styles';

export const Details = styled.p`
  margin: 0 20px;
  font-family: ${({ subtext }) => (subtext ? fontTitleAlt : fontSubtitle)};
  font-size: 0.9rem;
  line-height: 1;

  @media all and (min-width: 768px) {
    margin: 0 40px;
    font-size: 1rem;
    line-height: 1.3;
  }
  &:first-of-type {
    margin-top: 5px;
  }
`;
Details.displayName = 'Details';

const goIn = (formationItems, index) => keyframes`
  0% {
    margin-left : -100vw;
  }

  100% {
      margin-left: ${`${
        (10 / formationItems) * ((index + 1) * formationItems)
      }px`};  }
`;

export const DetailsArea = styled.div`
  margin-top: 10px;
  width: 80%;

  ${({ formationItems, index }) =>
    formationItems &&
    css`
      animation: ${goIn(formationItems, index)} 2s forwards;
    `}

  @media all and (min-width: 768px) {
    margin-top: 0;

    ${({ formationItems, index }) =>
      formationItems &&
      css`
        margin-left: ${`${
          (60 / formationItems) * ((index + 1) * formationItems)
        }px`};
      `}
  }
`;
DetailsArea.displayName = 'Details Area';

export const FakeText = styled.p`
  width: 100%;
  font-size: 0.9rem;
  line-height: 0.8;
  text-align: center;
`;
FakeText.displayName = 'Fake Text';

export const FormationArea = styled.div`
  display: flex;
  justify-content: flex-start;
  margin: 0 auto;
  height: 100%;
  width: 100%;
`;
FormationArea.displayName = 'Formation Area';

export const FormationItem = styled.div`
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
  margin: 20px 0;
  height: auto;
  width: 100%;
`;
FormationItem.displayName = 'Formation Item';

export const FormationTitle = styled(Title)`
  &:hover {
    letter-spacing: 0.1rem;
    color: ${colorYellowDark};
  }
`;
FormationTitle.displayName = 'Formation Title';

export const FormationWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-end;
  margin: 50px auto 0;
  padding-right: 5vw;
  height: ${`${window.innerHeight}px`};
  width: 100%;

  @media all and (min-width: 768px) {
    margin: 0 auto;
    height: auto;
    justify-content: space-around;
  }
`;
FormationWrapper.displayName = 'Formation Wrapper';

export const SectionFormation = styled.section`
  z-index: 6;

  ${({ visible }) =>
    visible &&
    css`
      opacity: 0.95;
      transform: none;
      height: 100%;
      width: 100vw;

      &:hover {
        border: 2px solid ${colorBlack};
        opacity: 0.95;
      }
    `}

  @media all and (min-width: 768px) {
    position: absolute;
    bottom: 50px;
    left: 44%;
    opacity: 0.5;
    transform: perspective(5em) rotateX(50deg);
    outline: none;
    border: 1px solid ${colorBlack};
    border-radius: 10px 0;
    height: 40px;
    width: 60px;
    overflow: hidden;
    cursor: pointer;
    box-shadow: 0 4px 5px 0 ${rgba(colorBlack, 0.4)},
      inset 0 1px 5px 1px ${rgba(colorWhite, 0.7)};
    background-image: linear-gradient(
      to top right,
      ${colorWhite},
      ${colorBlueWater},
      ${colorBlueLight}
    );

    &:hover {
      opacity: 0.7;
      border: 3px solid ${colorYellowBright};
    }

    ${({ visible }) =>
      visible &&
      css`
        top: 40px;
        left: 20vw;
        opacity: 0.95;
        transform: none;
        border: 2px solid ${colorBlack};
        border-radius: 40px 0;
        height: 90vh;
        width: 60vw;
        padding: 15px;
        cursor: default;

        &:hover {
          border: 2px solid ${colorBlack};
          opacity: 0.95;
        }
      `}
  }
`;
SectionFormation.displayName = 'FORMATION Section';

export const VerticalBar = styled.div`
  position: absolute;
  left: 10%;
  border-right: 5px solid ${rgba(colorBlueLight, 0.3)};
  height: 100%;
  width: 0;
`;
VerticalBar.displayName = 'Vertical Bar';
