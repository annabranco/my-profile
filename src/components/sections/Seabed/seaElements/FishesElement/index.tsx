import { ReactElement, useEffect } from 'react';
import { useStateWithLabel } from '../../../../../utils/hooks';
import { LEFT, RIGHT } from '../../../../../constants';
import { FISHES } from '../../../../../assets/images';
import { Fish } from './styles';

const randomSpeed = (num: number): string => `${num + 5}s`;
const randomSide = (num: number): string => (num === 0 ? RIGHT : LEFT);
const randomDelay = (num: number): string => `${num.toFixed(1)}s`;
const randomHeight = (num: number): string => {
  switch (num) {
    case 0:
      return '20%';
    case 1:
      return '30%';
    case 2:
      return '60%';
    case 3:
    default:
      return '70%';
  }
};
const randomMovement = (num: number): string => {
  switch (num) {
    case 0:
      return 'ease-in';
    case 1:
      return 'ease-out';
    case 2:
      return 'ease-in-out';
    case 3:
    default:
      return 'linear';
  }
};

const FishesElement = (): ReactElement => {
  const [fishes, updateFishes] = useStateWithLabel<ReactElement[]>(
    [],
    'fishes'
  );

  useEffect(() => {
    let currentFishes: ReactElement[] = [];
    const deployFish = (type: number, order: number): void => {
      currentFishes.push(
        <Fish
          alt="Swimming Fish" // TODO language
          bottom={randomHeight(Math.ceil(Math.random() * 4))}
          delay={randomDelay(Math.random() * 3)}
          facing={FISHES[type].facing}
          key={`fish${type}-${order}`}
          movementType={randomMovement(Math.floor(Math.random() * 4))}
          side={randomSide(Math.floor(Math.random() * 2))}
          speed={randomSpeed(Math.ceil(Math.random() * 5))}
          src={FISHES[type].img}
          waving={FISHES[type].waving || false}
        />
      );
    };

    for (let i = 1; i <= Math.ceil(Math.random() * 4); i++) {
      deployFish(Math.floor(Math.random() * FISHES.length), i);
    }

    updateFishes(currentFishes);

    setInterval(() => {
      currentFishes = [];
      for (let i = 1; i <= Math.ceil(Math.random() * 3); i++) {
        deployFish(Math.floor(Math.random() * FISHES.length), i);
      }
      updateFishes(currentFishes);
    }, 12000);
  }, [updateFishes]);

  return <>{fishes.map(fish => fish)}</>;
};

export default FishesElement;
