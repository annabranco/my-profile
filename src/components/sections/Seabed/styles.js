import styled, { css, keyframes } from 'styled-components';
import { rgba } from 'polished';
import { LEFT, ON_THE_LEFT } from '../../../constants';
import { Sea, SeaBed } from '../../../assets/images';
import { NotDisplayed } from '../../../styles/global';
import {
  colorBlack,
  colorBlueDark,
  colorWhite,
  colorYellowBright,
  colorYellowPale,
  fontTitle,
  colorGrayDark,
  fontTitleAlt,
  colorRedBright
} from '../../../styles/theme';

const FloatingAnimation = keyframes`
  from {
    transform : translate(-5px, 10px);
  }

  to {
    transform : translate(3px, -10px);
  }
`;

const Bubbling = keyframes`
  0%, 55%, 100% {
    bottom: 0;
    opacity: 0;
    transform:translateX(0);
  }
  3% {
    opacity: 0.5;
  }
  25% {
    transform:translateX(50px);
  }
  50% {
    bottom: ${`${window.innerHeight / 2.5}px`};
    transform:translateX(-50px);
  }
`;

const FloatingSoft = css`
  animation-name: ${FloatingAnimation};
  animation-duration: 7s;
  animation-timing-function: ease;
  animation-iteration-count: infinite;
  animation-direction: alternate;
  animation-fill-mode: both;
`;

const Floating = css`
  z-index: 5;
  animation-name: ${FloatingAnimation};
  animation-duration: 3s;
  animation-timing-function: ease;
  animation-iteration-count: infinite;
  animation-direction: alternate;
  animation-fill-mode: both;
`;

const FloatingDelay = css`
  animation-delay: 1s;
`;

const SeabedSubsectionContainer = styled.div`
  position: absolute;
  display: flex;
  width: 120px;
`;

export const Bubble = styled.div`
  position: absolute;
  bottom: 0;
  background-color: #f1f1f1;
  border-radius: 50%;
  opacity: 0;
  animation-name: ${Bubbling};
  animation-duration: 10s;
  animation-timing-function: ease-in-out;
  animation-iteration-count: infinite;

  &:nth-of-type(1) {
    left: 1%;
    height: 10px;
    width: 10px;
    animation-duration: 10s;
  }
  &:nth-of-type(2) {
    left: 10%;
    height: 12px;
    width: 12px;
    animation-duration: 5s;
    animation-delay: 0.3s;
  }
  &:nth-of-type(3) {
    left: 15%;
    height: 5px;
    width: 5px;
    animation-duration: 7s;
    animation-delay: 0.4s;
  }
  &:nth-of-type(4) {
    left: 30%;
    height: 10px;
    width: 10px;
    animation-duration: 10s;
    animation-delay: 0.5s;
  }
  &:nth-of-type(5) {
    left: 40%;
    height: 15px;
    width: 15px;
    animation-duration: 6s;
    animation-delay: 0.2s;
  }
  &:nth-of-type(6) {
    left: 50%;
    height: 5px;
    width: 5px;
    animation-duration: 7s;
    animation-delay: 0.1s;
  }
  &:nth-of-type(7) {
    left: 55%;
    height: 10px;
    width: 10px;
    animation-duration: 10s;
    animation-delay: 0.4s;
  }
  &:nth-of-type(8) {
    left: 60%;
    height: 7px;
    width: 7px;
    animation-duration: 6s;
    animation-delay: 0.5s;
  }
  &:nth-of-type(9) {
    left: 65%;
    height: 10px;
    width: 10px;
    animation-duration: 10s;
    animation-delay: 0.25s;
  }
  &:nth-of-type(10) {
    left: 70%;
    height: 4px;
    width: 4px;
    animation-duration: 8s;
    animation-delay: 0.3s;
  }
`;
Bubble.displayName = 'Bubble';

export const Bubbles = styled.div`
  position: absolute;
  top: 0;
  right: 10px;
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
  width: 100px;
  height: 20px;

  ${({ isSwimming }) =>
    isSwimming &&
    css`
      top: 25px;
    `}

  ${({ facing }) =>
    facing === LEFT &&
    css`
      right: unset;
      left: 10px;
    `}
`;
Bubbles.displayName = 'Bubbles';

export const FloorText = styled.p`
  position: absolute;
  bottom: 20px;
  left: 17vw;
  transform: perspective(7rem) rotateX(16deg) skewX(20deg);
  width: 80%;
  text-shadow: 0 0 3px ${colorYellowBright};
  font-family: ${fontTitle};
  font-size: 9rem;
  line-height: 0.8;
  font-weight: bold;

  ${({ hidden }) =>
    hidden &&
    css`
      display: hidden;
    `}

  @media all and (min-width: 1400px) {
    left: 22vw;
  }
`;
FloorText.displayName = 'Floor Text';

export const GoBackText = styled.p`
  margin-top: 30vh;
  text-shadow: 0 0 2px ${colorBlack};
  font-family: ${fontTitle};
  font-size: 3rem;
  color: ${colorWhite};
  text-align: center;
`;
GoBackText.displayName = 'Go Back Text';

export const HeroImage = styled.img`
  width: 200px;
  transition: all linear 1s;
  ${NotDisplayed};

  @media all and (min-width: 768px) {
    display: block;
  }

  ${({ back2Surface, crossingBorder, isGoingUp, isSwimming, facing }) => {
    if (crossingBorder) {
      return css`
        transition: none;
      `;
    }

    if (isSwimming) {
      return css`
        ${FloatingSoft}
      `;
    }

    if (isGoingUp && back2Surface) {
      return css`
        transform: ${facing === LEFT ? 'rotate(70deg)' : 'rotate(290deg)'};
      `;
    }

    if (isGoingUp) {
      return css`
        transform: ${facing === LEFT ? 'rotate(70deg)' : 'rotate(290deg)'};
      `;
    }

    if (facing === LEFT) {
      return css`
        transform: scaleX(-1);
      `;
    }

    return css`
      ${FloatingSoft}
    `;
  }}
`;
HeroImage.displayName = 'The Hero Image';

export const HeroWrapper = styled.div`
  position: absolute;
  top: unset;
  bottom: 40%;
  opacity: 0.6;
  transition: all linear 1s;

  ${({ crossingBorder }) => {
    if (crossingBorder) {
      return css`
        transition: none;
      `;
    }
    return null;
  }}

  ${({ back2Surface, isGoingUp }) => {
    if (isGoingUp && back2Surface) {
      return css`
        bottom: unset;
        top: -200px;
        transition: all 7s linear;
      `;
    }

    if (isGoingUp) {
      return css`
        bottom: unset;
        top: 40%;
        transition: all 7s linear;
      `;
    }
    return null;
  }}
`;
HeroWrapper.displayName = 'HeroWrapper';

export const Message = styled.p`
  margin: 15px 0;
  font-family: $font__title;
  font-size: 1.2rem;
  line-height: 1.8;
  color: ${colorBlueDark};
  text-align: justify;
`;
Message.displayName = 'Message';

export const MessageContainer = styled.div`
  border-radius: 80px 0;
  margin: 60px auto 0;
  background-color: rgba($white, 0.05);
  height: auto;
  width: 60vw;
  padding: 10px 50px;
  ${FloatingSoft};
`;
MessageContainer.displayName = 'Message Container';

export const MessageOnMobileDevices = styled(Message)`
  ${NotDisplayed};
`;
MessageOnMobileDevices.displayName = 'Message';

export const SeabedCloseButton = styled.button`
  z-index: 4;
  position: absolute;
  top: 10px;
  right: 5px;
  border: 1px solid $[colorRedDark];
  border-radius: 5px 0;
  background-image: linear-gradient(
    ${rgba(colorWhite, 0.2)},
    ${rgba(colorRedBright, 0.9)}
  );
  padding: 2px 5px;
  text-shadow: 0 0 1px ${rgba(colorGrayDark, 0.51)};
  font-family: ${fontTitleAlt};
  font-size: 0.7rem;
  color: ${colorBlack};
  cursor: pointer;
  box-shadow: 0 1px 2px 0 ${rgba(colorBlack, 0.4)},
    inset 0 1px 5px 1px ${rgba(colorWhite, 0.7)};

  @media all and (min-width: 321px) {
    border-radius: 8px 0;
    padding: 2px 10px;
    font-size: 1.2rem;
  }

  @media all and (min-width: 768px) {
    right: 10px;
    border-radius: 12px 0;
    padding: 3px 12px;
    font-size: 1.4rem;
  }

  &:hover {
    transform: translate(0, 1px);
    text-decoration: none;
    box-shadow: inset 0 2px 5px 1px ${rgba(colorBlack, 0.4)};
  }

  ${({ hidden }) =>
    hidden &&
    css`
      ${NotDisplayed}
    `}
`;
SeabedCloseButton.displayName = 'Close Button';

export const SeabedFloor = styled.div`
  position: absolute;
  bottom: 0;
  opacity: 0.05;
  background-image: ${`url(${SeaBed})`};
  background-position: bottom;
  background-size: 100% 220%;
  height: 50vh;
  width: 100%;
`;
SeabedFloor.displayName = 'Seabed Floor';

export const SeabedSection = styled.section`
  position: relative;
  background-image: linear-gradient(
      ${rgba('#bdd6f6', 0.96)},
      ${rgba('#abccf6', 0.95)},
      ${rgba('#ecf6bd', 0.9)}
    ),
    ${`url(${Sea})`};
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  height: 100vh;
  width: 100%;
`;
SeabedSection.displayName = 'Seabed Section';

export const SubsectionFormation = styled(SeabedSubsectionContainer)`
  bottom: 150px;
  left: 20vw;
  justify-content: flex-start;
  text-align: center;
  ${Floating};
  ${FloatingDelay};

  @media all and (min-width: 768px) {
    bottom: 33vh;
    left: auto;
    right: 30px;
    justify-content: flex-end;
    text-align: right;
  }
  ${({ hidden }) =>
    hidden &&
    css`
      display: hidden;
    `}
`;
SubsectionFormation.displayName = 'Subsection Formation';

export const SubsectionOtherSkills = styled(SeabedSubsectionContainer)`
  top: 170px;
  bottom: initial;
  right: 33vw;
  justify-content: flex-end;
  text-align: center;
  ${Floating};

  @media all and (min-width: 768px) {
    top: initial;
    bottom: 33vh;
    left: 40px;
    right: auto;
    justify-content: flex-start;
    text-align: left;
  }
  ${({ hidden }) =>
    hidden &&
    css`
      display: hidden;
    `}
`;
SubsectionOtherSkills.displayName = 'Subsection Other Skills';

export const Text = styled.p`
  opacity: 0.9;
  text-shadow: 0 0 3px ${colorBlueDark};
  font-family: sans-serif;
  font-size: 2.8rem;
  color: ${colorWhite};

  @media all and (min-width: 768px) {
    opacity: 0.7;
    text-shadow: 0 0 3px ${colorWhite};
    font-size: 3rem;
    color: ${rgba(colorWhite, 0.2)};
    ${({ active }) =>
      active &&
      css`
        @media all and (min-width: 768px) {
          text-shadow: 0 0 3px ${colorWhite};
          font-size: 3.2rem;
          color: ${rgba(colorYellowBright, 0.73)};
        }
      `}
  }
`;
Text.displayName = 'Text';

export const TextFindSomething = styled.p`
  position: absolute;
  top: 20vh;
  left: 0;
  transform: translate(-50%, 0);
  width: 100%;
  text-shadow: 0 0 2px ${colorBlack};
  font-family: ${fontTitle};
  font-size: 2rem;
  color: ${colorWhite};
  text-align: center;

  @media all and (min-width: 768px) {
    top: 30vh;
    left: 50%;
    transform: translate(-50%, 0);
    font-size: 2.5rem;

    &:nth-of-type(2) {
      top: 37vh;
    }

    &:nth-of-type(3) {
      top: 44vh;
    }
  }

  &:nth-of-type(2) {
    top: 38vh;
  }
`;
TextFindSomething.displayName = 'Text Find Something';

export const ThinkingText = styled.p`
  position: absolute;
  bottom: 68%;
  opacity: 0.8;
  width: 180px;
  text-shadow: -1px 1px 6px ${colorBlack};
  font-style: italic;
  font-family: sans-serif;
  font-size: 1.2rem;
  color: ${colorYellowPale};
  text-align: center;

  ${({ side }) => {
    let leftValue = 'initial';
    let rightValue = '40px';

    if (side === ON_THE_LEFT) {
      leftValue = '40px';
      rightValue = 'initial';
    }
    return css`
      left: ${leftValue};
      right: ${rightValue};
    `;
  }}
`;
ThinkingText.displayName = 'Thinking Text';
