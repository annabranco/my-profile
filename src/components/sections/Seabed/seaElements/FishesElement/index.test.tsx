import React, { Component, ReactElement } from 'react';
import { mount, ReactWrapper, shallow, ShallowWrapper } from 'enzyme';
import 'jest-styled-components';
import FishesElement from '.';
import { Fish } from './styles';
import { LEFT } from '../../../../../constants';

// TODO [10-May -21]: Implement defined testing suite (Anna Branco)

const MockFish = () => (
  <Fish
    alt="Mocked Fish"
    bottom="20%"
    delay="0s"
    facing={LEFT}
    key="fish"
    movementType="ease-in"
    side={LEFT}
    speed="1s"
    src="fish"
    waving={false}
  />
);

describe('< FishesElement >', () => {
  let wrapper: ReactWrapper;

  beforeAll(() => {
    wrapper = mount(<FishesElement />);
  });

  it('should mount', () => {
    expect(wrapper).toBeDefined();
  });

  xit('matches the snapshot', () => {
    // As Fishes elements are created randomly on init, snapshots will never match
    expect(wrapper).toMatchSnapshot();
  });

  it('should render at least one Fish on screen', () => {
    jest.spyOn(React, 'useEffect').mockImplementation(f => f());

    expect(wrapper.find('Fish').length).toBeGreaterThanOrEqual(1);
  });
});
