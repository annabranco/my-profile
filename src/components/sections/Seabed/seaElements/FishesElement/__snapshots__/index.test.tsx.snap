// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< FishesElement > matches the snapshot 1`] = `
<FishesElement>
  <Fish
    alt="Swimming Fish"
    bottom="30%"
    delay="3.0s"
    facing="right"
    key="fish3-1"
    movementType="ease-in"
    side="left"
    speed="7s"
    src={
      Object {
        "process": [Function],
      }
    }
    waving={true}
  >
    <img
      alt="Swimming Fish"
      className="kRUVMb"
      speed="7s"
      src={
        Object {
          "process": [Function],
        }
      }
    />
  </Fish>
  <Fish
    alt="Swimming Fish"
    bottom="60%"
    delay="1.7s"
    facing="left"
    key="fish0-2"
    movementType="ease-in"
    side="right"
    speed="7s"
    src={
      Object {
        "process": [Function],
      }
    }
    waving={false}
  >
    <img
      alt="Swimming Fish"
      className="iAugPx"
      speed="7s"
      src={
        Object {
          "process": [Function],
        }
      }
    />
  </Fish>
</FishesElement>
`;
